dev/devicelab/lib/framework/cocoon.dart

```diff
@@ -64,26 +61,20 @@ class Cocoon {
     return _commitSha = result.stdout as String;
   }
 
-  /// Parse the local repo for the current running branch.
-  String _readCommitBranch() {
-    final ProcessResult result = processRunSync('git', <String>['rev-parse', '--abbrev-ref', 'HEAD']);
-    if (result.exitCode != 0) {
-      throw CocoonException(result.stderr as String);
-    }
-
-    return _commitBranch = result.stdout as String;
-  }
-
   /// Send [TaskResult] to Cocoon.
-  Future<void> sendTaskResult({String builderName, TaskResult result}) async {
+  Future<void> sendTaskResult({@required String builderName, @required TaskResult result, @required String gitBranch}) async {
+    assert(builderName != null);
+    assert(gitBranch != null);
```

Comment:
1) does this run with asserts enabled?
2) in what order do we land this (and the related PRs)?

Reply:
1. The tests will, which checks the calls to this . However, runtime issues can still happen regarding flags. I was under the assumption NNBD was being used by the test runner, is it not?
2. This lands first, then the recipes CL
3. No. It's only used on LUCI runs. Cocoon agent handles this logic in the agent

Reply:
2. what will the value of `gitBranch` be if it is not passed as a command-line option? My concern is just that we fail with a helpful error if somehow this method gets invoked with an invalid `gitBranch` value.

Reply:
shall we have `gitBranch` with a default value like HEAD?

Reply:
Oh, the `HEAD` causes the error.. nvm..

Reply:
I moved the error checking into cocoon with https://github.com/flutter/cocoon/pull/1022

This will give a more descriptive error than we can do in the test runner.